src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -876,9 +883,27 @@ internal void AddCluster(ClusterResultItem ClusterResultItem)
             });
 
             // Connect the cluster to the original node and port
-            var connector = ConnectorModel.Make(node.NodeModel, targetNodeFromCluster.NodeModel, 0, ClusterResultItem.EntryNodeInPort);
-            newNodesAndWires.Add(connector);
+            if (targetNodeFromCluster.InPorts.Any())
+            {
+                dynamoViewModel.Model.ExecuteCommand(new DynamoModel.MakeConnectionCommand(node.NodeModel.GUID, 0,
+                    PortType.Output, DynamoModel.MakeConnectionCommand.Mode.Begin));
+                dynamoViewModel.Model.ExecuteCommand(new DynamoModel.MakeConnectionCommand(targetNodeFromCluster.NodeModel.GUID, ClusterResultItem.EntryNodeInPort,
+                    PortType.Input, DynamoModel.MakeConnectionCommand.Mode.End));
+
+                var lastConnector = wsViewModel.Connectors.Last();
```

Comment:
perhaps more null checking ; with a cluster that consists of a single node and also a single node in the graph we might have 0 connectors when our command fails.